pkg/termination/termination_test.go

```diff
@@ -0,0 +1,354 @@
+		})
+
+		Context("and the poll URL cannot be reached", func() {
+			BeforeEach(func() {
+				h.pollURL = &url.URL{Opaque: "abc#1://localhost"}
+			})
+
+			It("should return an error", func() {
+				Eventually(errs).Should(Receive(MatchError("error polling termination endpoint: could not get URL \"abc#1://localhost\": Get abc#1://localhost: unsupported protocol scheme \"\"")))
+			})
+
+			It("should not delete the machine", func() {
+				key := client.ObjectKey{Namespace: testMachine.Namespace, Name: testMachine.Name}
+				Consistently(func() error {
+					m := &machinev1.Machine{}
+					return k8sClient.Get(ctx, key, m)
+				}).Should(Succeed())
+			})
+		})
+	})
+
+	Context("getMachineForNode", func() {
+		var machine *machinev1.Machine
+		var err error
+
+		JustBeforeEach(func() {
+			machine, err = h.getMachineForNode(ctx)
+		})
+
+		Context("with a broken client", func() {
+			BeforeEach(func() {
+				brokenClient, err := client.New(cfg, client.Options{Scheme: runtime.NewScheme()})
+				Expect(err).ToNot(HaveOccurred())
+				h.client = brokenClient
+			})
+
+			It("should return an error", func() {
+				Expect(err).ToNot(BeNil())
+				Expect(err.Error()).To(HavePrefix("error listing machines: no kind is registered for the type v1beta1.MachineList in scheme"))
+			})
+
+			It("should not return a machine", func() {
+				Expect(machine).To(BeNil())
+			})
+		})
+
+		Context("with no machine for the node name", func() {
+			It("should return an error", func() {
+				Expect(err).To(MatchError("machine not found for node \"test-node\""))
+			})
+
+			It("should not return a machine", func() {
+				Expect(machine).To(BeNil())
+			})
+		})
+
+		Context("with a machine matching the node name", func() {
+			var testMachine *machinev1.Machine
+
+			BeforeEach(func() {
+				testMachine = newTestMachine("test-machine", testNamespace, nodeName)
+				createMachine(testMachine)
+			})
+
+			It("should not return an error", func() {
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			It("should return a machine", func() {
+				Expect(machine).To(Equal(testMachine))
+			})
+		})
+	})
+})
+
+var _ = Describe("ScheduledEvents Suite", func() {
+	Context("ummarshalling from json", func() {
+		scheduledEvents := &scheduledEvents{}
+		jsonData := []byte(`
+    {
+      "DocumentIncarnation":0,
+      "Events": [
+                {
+                  "EventType":"Preempt", 
+                  "ResourceType": "VirtualMachine"
+                },
+                {
+                  "EventType":"Preempt", 
+                  "ResourceType": "Terminate"
+                }
+      ]
+    }`)
+
+		It("json unmarshal shouldn't fail", func() {
+			Expect(json.Unmarshal(jsonData, scheduledEvents)).To(Succeed())
+		})
+
+		It("number of events to be equal to 2", func() {
+			Expect(len(scheduledEvents.Events)).To(Equal(2))
+		})
+
+		It("first event to have event type Preempt", func() {
+			Expect(scheduledEvents.Events[0].EventType).To(Equal(preemptEventType))
+		})
```

Comment:
This relies on test ordering, you should refactor this so that the unmarshal is done in a before each

```
Context("unmarshalling from json", func() {
  var scheduledEvents *scheduledEvents
  jsonData := ...

  BeforeEach(func() {
    scheduleEvents = &scheduledEvents{}
    Expect(json.Unmarshal(jsonData, scheduledEvents)).To(Succeed())
  })

  It(...)
})
```